lib/ansible/plugins/connection/docker.py

```diff
@@ -205,10 +207,55 @@ def exec_command(self, cmd, in_data=None, sudoable=False):
 
         display.vvv("EXEC %s" % (local_cmd,), host=self._play_context.remote_addr)
         local_cmd = [to_bytes(i, errors='surrogate_or_strict') for i in local_cmd]
-        p = subprocess.Popen(local_cmd, shell=False, stdin=subprocess.PIPE,
-                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
 
+        p = subprocess.Popen(
+            local_cmd,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+
+        if self.become and self.become.expect_prompt() and sudoable:
+            display.debug("handling privilege escalation")
+            fcntl.fcntl(p.stdout, fcntl.F_SETFL, fcntl.fcntl(p.stdout, fcntl.F_GETFL) | os.O_NONBLOCK)
+            fcntl.fcntl(p.stderr, fcntl.F_SETFL, fcntl.fcntl(p.stderr, fcntl.F_GETFL) | os.O_NONBLOCK)
+
+            selector = selectors.DefaultSelector()
+            selector.register(p.stdout, selectors.EVENT_READ)
+            selector.register(p.stderr, selectors.EVENT_READ)
+
+            become_output = b''
+            try:
+                while not self.become.check_success(become_output) and not self.become.check_password_prompt(become_output):
+                    events = selector.select(self._play_context.timeout)
+                    if not events:
+                        stdout, stderr = p.communicate()
+                        raise AnsibleError('timeout waiting for privilege escalation password prompt:\n' + to_native(become_output))
+
+                    for key, event in events:
+                        if key.fileobj == p.stdout:
+                            chunk = p.stdout.read()
+                            break
```

Comment:
I don't think this `break` should be there (and I guess I missed it in the original PR). It's not in `local.py`, either. @larsks where does it come from?

Reply:
Huh, I haven't the foggiest idea. I didn't do any editing of that content that I recall.  Funny that it works.  I guess we submit a patch and fix it in the backports? Funny that everything still works.  I wish this was all handled in one place...

Reply:
I've created #56288 and will include it in this backport once it is merged.

Your stable-2.7 does not have this code included though...